### PR TITLE
Add how to use as module to the README

### LINKED ISSUE #148: Examples for using as a library
Love this project. Can we get API docs for using yapf as a library please?

### PR DESCRIPTION
**Pull request by hayd**
Fix #148.

Note: IMO the FormatFile api is a little strange atm, in returning (source, encoding) or (diff, encoding). I think an in_place argument would be good... but I'm not sure what it ought to return in this case ([None](https://github.com/google/yapf/blob/25de78025392360224e7ec40a90c1502377e34d5/yapf/yapflib/yapf_api.py#L71), encoding) ?

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -175,6 +175,65 @@ and reformat it into:
...
+You can also pass a ``style_config``: Either a style name or a path to a file that contains
+formatting style settings. If None is specified, use the default style
+as set in ``style.DEFAULT_STYLE_FACTORY``.
+
+.. code-block:: python
+
+    >>> FormatCode("def g():\n  return True", style_config='pep8')
+    'def g():\n    return True\n'
+
+    >>> FormatFile("def g():\n  return True", style_config='pep8')
```
Comment (hayd) [inline README.rst]: eurgh, this is a typo. Doh!

**Comment by gwelymernans** [comment]
Oops! I didn't catch this. I mentioned in the other pull request that it should probably return `None, encoding`. That seems the most logical thing to do, IMHO. But I could be wrong. :-)

File: README.rst (hunk 1 above)
Comment (gwelymernans) [inline README.rst]: Go ahead and fix it and then squash the result. :-)

**Comment by hayd** [comment]
I may cheekily wait for #158 and then update to include mention of in_place! :)

Like I said there, having looked a little, I'm not *sure* that (None, encoding) makes sense, because then you have the same result whether:

- the file exists, is reformatted, writes out sucessfully
- the file does not exist, and exits\*

\* for some reason currently this returns (None, encoding) rather than raising an exception of some sort... :s

**Pull request merged**